### PR TITLE
Fix select component animation

### PR DESCRIPTION
**Pull request by Tankenstein**
This adds support for animations in `Select` again by not unmounting the dropdown right away. We could've just kept it hidden, but i also used a `Transition` to unmount it after the animation is done.
Also refactors some things in docs and building classnames in Select.

### CODE REVIEW

File: docs/Docs.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ import DefinitionListDocs from './DefinitionListDocs';
 import UploadDocs from './UploadDocs';
 
 const Docs = () => (
-  <Fragment>
+  <StrictMode>
```
Comment (andreapiras) [inline docs/Docs.js]: image
We need to either fix those errors or open a ticket to fix them :)

File: src/select/Select.js [hunk 2]
```diff
@@ -23,6 +25,10 @@ function stopPropagation(event) {
   // document listener does not use SyntheticEvents
 }
 
+const BOOTSTRAP_DROPDOWN_ANIMATION_TIME = 200;
+
+const defer = fn => setTimeout(fn, 0);
+
 export default class Select extends Component {
```
Comment (andreapiras) [inline src/select/Select.js]: This can probably be change too to PureComponent

File: src/select/Select.js [hunk 3]
```diff
@@ -345,64 +351,93 @@ export default class Select extends Component {
...
-
-    const dropdownMenu = this.style('dropdown-menu');
-    const dropdownRight = this.props.dropdownRight
-      ? this.style(`dropdown-menu-${this.props.dropdownRight}-right`)
-      : '';
-    const dropdownWidth = this.props.dropdownWidth
-      ? this.style(`dropdown-menu-${this.props.dropdownWidth}`)
-      : '';
-    const dropdownClass = `${dropdownMenu} ${dropdownRight} ${dropdownWidth}`;
+    const s = this.style;
```
Comment (andreapiras) [inline src/select/Select.js]: Personal preference but I would put a more descriptive name here

File: src/select/Select.js [hunk 4]
```diff
@@ -345,64 +351,93 @@ export default class Select extends Component {
...
-            {!required && !canSearch ? this.renderPlaceHolderOption() : ''}
-            {canSearch ? this.renderSearchBox() : ''}
-            {this.renderOptions()}
-          </ul>
+        {animationState => (
+          <div // eslint-disable-line jsx-a11y/no-static-element-interactions
+            className={groupClass}
+            ref={this.dropdownMenuRef}
+            onKeyDown={this.handleKeyDown}
+            onTouchMove={this.handleTouchStart}
```
Comment (andreapiras) [inline src/select/Select.js]: should this be handleTouchMove?

File: src/select/Select.js [hunk 5]
```diff
@@ -345,64 +351,93 @@ export default class Select extends Component {
...
+            <button
+              disabled={disabled}
+              className={buttonClass}
+              type="button"
+              id={id}
+              aria-expanded={open}
+              onClick={this.handleButtonClick}
+            >
+              {this.renderButtonInternals()}
+              <span className={this.style('caret')} />
```
Comment (andreapiras) [inline src/select/Select.js]: s?

File: docs/Docs.js (hunk 1 above)
Comment (Tankenstein) [reply docs/Docs.js]: Doesn't block this PR though :)

File: src/select/Select.js (hunk 2 above)
Comment (Tankenstein) [reply src/select/Select.js]: No it can't, referential equality will not work for `options`

File: src/select/Select.js (hunk 3 above)
Comment (Tankenstein) [reply src/select/Select.js]: This was for shortening all the stuff below 😅 am fine to change it back

File: docs/Header.js [hunk 6]
```diff
@@ -1,53 +1,21 @@
...
+    <section className="section row">
+      <div className="col-md-6">
+        <h1>
+          Components{' '}
+          <span role="img" aria-label="none">
+            🎉
+          </span>{' '}
+          <small>v{npmPackage.version}</small>
+        </h1>
+        <p>TransferWise design system components in react.</p>
```
Comment (dozortsev) [inline docs/Header.js]: should we mention that's called Neptune?)

File: docs/Docs.js (hunk 1 above)
Comment (andreapiras) [reply docs/Docs.js]: Correct but would be better to have them addressed in a ticket :)

Comment (Tankenstein) [reply docs/Docs.js]: Good idea 👍  https://transferwise.atlassian.net/jira/software/projects/DS/boards/277?selectedIssue=NP-49

**Pull request merged**